ps/jenkins/package-testing-ps-build-5.7.groovy

```diff
@@ -0,0 +1,230 @@
+library changelog: false, identifier: 'lib@master', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/Percona-Lab/jenkins-pipelines.git'
+]) _
+
+setup_rhel_package_tests = { ->
+    sh '''
+        sudo yum install -y epel-release
+        sudo yum -y update
+        sudo yum install -y ansible git wget
+    '''
+}
+
+setup_amazon_package_tests = { ->
+    sh '''
+        sudo amazon-linux-extras install epel
+        sudo yum -y update
+        sudo yum install -y ansible git wget
+    '''
+}
+
+setup_debian_package_tests = { ->
+    sh '''
+        sudo apt-get install -y dirmngr gnupg2
+        echo "deb http://ppa.launchpad.net/ansible/ansible/ubuntu trusty main" | sudo tee -a /etc/apt/sources.list > /dev/null
```

Comment:
why do you need trusty repo? trusty is EOL

Reply:
The installation instructions for Ansible on Debian seem to recommend using their Trusty PPA: https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#installing-ansible-on-debian

Reply:
```
vagrant@ubuntu-bionic:~$ lsb_release -c
Codename:       bionic
vagrant@ubuntu-bionic:~$ apt-cache search ansible
ansible - Configuration management, deployment, and task execution system
ansible-lint - lint tool for Ansible playbooks
ansible-tower-cli - command line tool for Ansible Tower and AWX Project
ansible-tower-cli-doc - documentation for tower-cli command line tool and library
bootstrap-vz - tool for creating Debian images for cloud platforms (CLI)
pyinfra - state based and programmable service deployment tool
python-reclass - hierarchical inventory backend for configuration management systems
python-tower-cli - Python 2 client library for the Ansible Tower and AWX Project's REST API
python3-tower-cli - Python 3 client library for the Ansible Tower and AWX Project's REST API
reclass - hierarchical inventory backend for configuration management systems
reclass-doc - reclass documentation
ssg-applications - SCAP Guides and benchmarks targeting userspace applications
ssg-debderived - SCAP Guides and benchmarks targeting Debian-based OS
ssg-debian - SCAP Guides and benchmarks targeting Debian 8
ssg-nondebian - SCAP Guides and benchmarks targeting other GNU/Linux OS
vim-syntastic - Syntax checking hacks for vim

```
So seems the mentioned doc is outdated 
Please check how ansible could be installed on our supported platforms and enable outdated repos only on platforms where there is no other way to install ansible

Reply:
on debian 10:
```
vagrant@debian-10:~$ sudo apt-cache search ansible
ansible - Configuration management, deployment, and task execution system
ansible-doc - Ansible documentation and examples
ansible-lint - lint tool for Ansible playbooks
ansible-tower-cli - command line tool for Ansible Tower and AWX Project
ansible-tower-cli-doc - documentation for tower-cli command line tool and library
python-tower-cli - Python 2 client library for the Ansible Tower and AWX Project's REST API
python3-tower-cli - Python 3 client library for the Ansible Tower and AWX Project's REST API
bootstrap-vz - tool for creating Debian images for cloud platforms (CLI)
shade-inventory - Ansible inventory script for OpenStack clouds
python-reclass - hierarchical inventory backend for configuration management systems
reclass - hierarchical inventory backend for configuration management systems
reclass-doc - reclass documentation
ssg-applications - SCAP Guides and benchmarks targeting userspace applications
ssg-debderived - SCAP Guides and benchmarks targeting Debian-based OS
ssg-debian - SCAP Guides and benchmarks targeting Debian 8
ssg-nondebian - SCAP Guides and benchmarks targeting other GNU/Linux OS
vim-syntastic - Syntax checking hacks for vim

```

Reply:
While, as you mentioned, Trusty itself is EOL, the Ansible PPA for Trusty is still getting updates regularly, last one (2.9.22) as of last week: https://launchpad.net/~ansible/+archive/ubuntu/ansible

And while both Debian Stretch and Buster ship Ansible on their main repositories, they ship older versions (2.2.1 for Stretch and 2.7.7 for Buster), which is probably why Ansible recommends using their own PPAs instead (not sure why they recommend Trusty specifically, but it's as up-to-date as the others).

I could change it so that Buster uses 2.7.7 from the Debian repos instead, but I can't make the same change for Stretch as our playbooks don't run on 2.2.1. Regardless, I think there's a benefit in using the same version of Ansible across builds, and the Ansible PPA doesn't seem to be outdated.